src/Paprika.Benchmarks/SlottedArrayBenchmarks.cs

```diff
@@ -191,6 +246,59 @@ public int EnumerateNibble(byte nibble)
         return length;
     }
 
+    [Benchmark]
+    public bool Defragmentation()
+    {
+        var map = CreateMapForDefragmentation();
+
+        // Insert additional key to trigger defragmentation
+        var additionalKey = Keccak.Zero;
+        Span<byte> value = stackalloc byte[32];
+
+        for (byte i = 0; i < 32; i++)
+        {
+            value[i] = i;
+        }
+
+        return map.TrySet(NibblePath.FromKey(additionalKey), value);
+    }
+
+    [Benchmark]
+    [Arguments(1)]
+    [Arguments(25)]
+    [Arguments(50)]
+    [Arguments(DefragmentationKeyCount - 1)]
+    public bool Defragmentation_Large(int count)
+    {
+        var map = CreateMapForDefragmentation();
+
+        // Delete random keys
+        for (var i = 0; i < count; i++)
+        {
+            map.Delete(GetRandomDefragKey((byte)i, false));
+        }
+
+        // Insert additional key to trigger defragmentation
+        var additionalKey = Keccak.Zero;
+        Span<byte> value = stackalloc byte[32];
+
+        for (byte i = 0; i < 32; i++)
+        {
+            value[i] = i;
+        }
+
+        return map.TrySet(NibblePath.FromKey(additionalKey), value);
```

Comment:
This sets just one value, right? What about the cost of the creating the map? Should `Defrag...` benchmark be used as a baseline?